front/src/applications/operationalStudies/views/Scenario.tsx

```diff
@@ -97,25 +100,45 @@ export default function Scenario() {
     }
   }, [infra]);
 
+  useEffect(() => {
+    if (infra && infra.state === 'CACHED' && timetableId) {
+      getTimetableWithTrainSchedulesDetails({ id: timetableId })
+        .unwrap()
+        .then((result) => {
+          getTimetable(result);
+        });
+
+      getTimetableConflicts({ id: timetableId })
+        .unwrap()
+        .then((data) => {
+          setConflicts(data as Conflict[]);
+        });
+    }
+  }, [infra, timetableId]);
+
+  useEffect(() => {
+    if (timetableId && reloadTimetable) {
+      getTimetableConflicts({ id: timetableId })
+        .unwrap()
+        .then((data) => {
+          setConflicts(data as Conflict[]);
+        });
+    }
+  }, [reloadTimetable]);
+
+  useEffect(() => {
+    if (infraId) {
+      reloadInfra({ id: infraId as number }).unwrap();
+    }
+  }, [infraId]);
+
```

Comment:
Same here, it's not possible to make it more concise ? less useEffect ! :D

Reply:
Yeah, I'm not a fan either for these 3 but : 
- If i put the 3rd one in the 1st one. It means each time infra will change (and during infra load, polling refetch infra every second so it changes almost every second), it will relaunch the reloadInfra so we will have 2 calls (the reload and the polling) instead of one. The 3rd is actually just here when we enter the page to check if infra is still up to date or if it need to be reloaded.
- If i put the 2nd one in the 1st one, it means each time reloadTimetalbe will change (it always goes from false > true > false when adding/deleting/modifying a train), it will launch multiple api calls for both endpoint in the first useEffect.

If you have a better suggestion/workflow for that, I'd be very happy to try it :)

Reply:
Probably we will be able to remove the 2nd one when using rtk query tags